lib/run-datadog.sh

```diff
@@ -79,35 +80,47 @@ start_datadog() {
     # Create folder for storing PID files
     mkdir run
 
-    # DSD requires its own config file
-    cp $DATADOG_DIR/dist/datadog.yaml $DATADOG_DIR/dist/dogstatsd.yaml
-    if [ -n "$RUN_AGENT" -a -f ./agent ]; then
-      if [ "$DD_LOGS_ENABLED" = "true" -a "$DD_LOGS_VALID_ENDPOINT" = "false" ]; then
-        echo "Log endpoint not valid, not starting agent"
-      else
-        export DD_LOG_FILE=$DATADOG_DIR/agent.log
-        export DD_IOT_HOST=false
-        sed -i "s~log_file: AGENT_LOG_FILE~log_file: $DD_LOG_FILE~" $DATADOG_DIR/dist/datadog.yaml
-        if [ "$SUPPRESS_DD_AGENT_OUTPUT" == "true" ]; then
-          ./agent run --cfgpath $DATADOG_DIR/dist/ --pidfile $DATADOG_DIR/run/agent.pid > /dev/null 2>&1 &
-        else
-          ./agent run --cfgpath $DATADOG_DIR/dist/ --pidfile $DATADOG_DIR/run/agent.pid &
-        fi
-      fi
-    else
-      export DD_LOG_FILE=$DATADOG_DIR/dogstatsd.log
+
+    # Run agent when checks are enabled
+    if [ "$DD_ENABLE_CHECKS" = "true" -a -f ./agent ]; then
+      export DD_LOG_FILE=$DATADOG_DIR/agent.log
+      export DD_IOT_HOST=false
       sed -i "s~log_file: AGENT_LOG_FILE~log_file: $DD_LOG_FILE~" $DATADOG_DIR/dist/datadog.yaml
-      if [ "$SUPPRESS_DD_AGENT_OUTPUT" == "true" ]; then
-        ./dogstatsd start --cfgpath $DATADOG_DIR/dist/ > /dev/null 2>&1 &
+      if [ "$SUPPRESS_DD_AGENT_OUTPUT" = "true" ]; then
+        ./agent run --cfgpath $DATADOG_DIR/dist/ --pidfile $DATADOG_DIR/run/agent.pid > /dev/null 2>&1 &
       else
-        ./dogstatsd start --cfgpath $DATADOG_DIR/dist/ &
+        ./agent run --cfgpath $DATADOG_DIR/dist/ --pidfile $DATADOG_DIR/run/agent.pid &
       fi
-      echo $! > run/dogstatsd.pid
     fi
-    if [ "$SUPPRESS_DD_AGENT_OUTPUT" == "true" ]; then
-      ./trace-agent --config $DATADOG_DIR/dist/datadog.yaml --pid $DATADOG_DIR/run/trace-agent.pid > /dev/null 2>&1 &
+
+    # DSD requires its own config file
+    cp $DATADOG_DIR/dist/datadog.yaml $DATADOG_DIR/dist/dogstatsd.yaml
+
+    # trace agent own config file
+    cp $DATADOG_DIR/dist/datadog.yaml $DATADOG_DIR/dist/datadog_trace.yaml
+
+    # set the trace agent and dogstatsd hostnames to the VM hostname
+    host $CF_INSTANCE_IP
+    if [ $? -eq 0 ]; then
+        IFS=. read -a VM_HOSTNAME <<< $(host $CF_INSTANCE_IP | awk '{print $5}')
+        sed -i "s~# hostname: mymachine.mydomain~hostname: $VM_HOSTNAME~" $DATADOG_DIR/dist/datadog_trace.yaml
+        sed -i "s~# hostname: mymachine.mydomain~hostname: $VM_HOSTNAME~" $DATADOG_DIR/dist/dogstatsd.yaml
```

Comment:
I would simplify this and just append the hostname to the config file, since this is already a comment, it can stay there (or even remove it directly from the packaged "template" config file)